gax/src/main/java/com/google/api/gax/rpc/StubSettings.java

```diff
@@ -90,10 +93,16 @@ protected StubSettings(Builder builder) {
     this.tracerFactory = builder.tracerFactory;
   }
 
+  /** @deprecated Please use {@link #getWorkerExecutorProvider()}. */
+  @Deprecated
   public final ExecutorProvider getExecutorProvider() {
     return executorProvider;
   }
 
+  public final ExecutorProvider getWorkerExecutorProvider() {
```

Comment:
I dont love the name workerExecutor...it doesnt really explain do it does. It's purpose is to schedule retries and batch flushes. gRPC calls it a [timer service](
https://github.com/grpc/grpc-java/blob/42d7fba1b86ad711687944dc91d0ea8ab4908f44/core/src/main/java/io/grpc/internal/GrpcUtil.java#L549).

Which is better but not not perfect. Other imperfect names that come to mind:
- backgroundExecutor
- scheduler
wdyt?

Reply:
backgroundExecutor sounds good